### PR TITLE
Add metrics for syncer states

### PR DESCRIPTION
**Pull request by sawsa307**
Added metrics to collect the current count of syncers in each status.

### CODE REVIEW

File: pkg/neg/metrics/neg_metrics_collector.go [hunk 1]
```diff
@@ -39,6 +42,16 @@ var (
 		},
 		[]string{syncResultLabel},
 	)
+
+	// syncerSyncerStatus tracks the count of syncer in different statuses
```
Comment (swetharepakula) [inline pkg/neg/metrics/neg_metrics_collector.go:46]: nit: different states

**Review by swetharepakula — COMMENTED** [review]
small nit, otherwise ready to be merged

File: pkg/neg/metrics/neg_metrics_collector.go (hunk 1 above)
Comment (sawsa307) [reply pkg/neg/metrics/neg_metrics_collector.go]: Updated. Thanks!

**Review by swetharepakula — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**